source/NPC.cpp

```diff
@@ -131,6 +137,61 @@ void NPC::Load(const DataNode &node)
 			else
 				location.Load(child);
 		}
+		else if(child.Token(0) == "waypoint")
+		{
+			// Waypoints can be added one-by-one (e.g. if from LocationFilters) or
+			// from multiple "waypoint" nodes. If no system or filter to pick a system is
+			// passed to the node, the mission's destination system will be used.
+			if(!child.HasChildren())
+			{
+				// Given "waypoint" or "waypoint <system 1> .... <system N>"
+				if(child.Size() == 1)
+					missingWaypoint = true;
```

Comment:
what if there was a waypoint node before this one that added something?

Reply:
Hmm... I'm not sure. I _think_ it would behave normally, though, since it only checks once if there's a missing waypoint in `Instantiate` and adds the mission destination if there is.

Reply:
ah ok ye works